src/cmd_vel_mux.cpp

```diff
@@ -79,144 +82,253 @@ CmdVelMux::CmdVelMux(rclcpp::NodeOptions options) : rclcpp::Node("cmd_vel_mux",
 }
 
-void CmdVelMux::configureFromParameters(const std::vector<std::string> & names, const std::vector<std::string> & topics, const std::vector<double> & timeouts, const std::vector<int64_t> & priorities, const std::vector<std::string> & short_descs)
+bool CmdVelMux::parametersAreValid(const std::map<std::string, ParameterValues> & parameters) const
+{
+  for (const std::pair<std::string, ParameterValues> & parameter : parameters)
+  {
+    if (parameter.second.topic.empty())
+    {
+      RCLCPP_WARN(get_logger(), "Empty topic for '%s'", parameter.first.c_str());
+      return false;
+    }
+    if (parameter.second.timeout < 0.0)
+    {
+      RCLCPP_WARN(get_logger(), "Missing timeout for '%s', ignoring", parameter.first.c_str());
+      return false;
+    }
+    if (parameter.second.priority < 0)
+    {
+      RCLCPP_WARN(get_logger(), "Missing priority for '%s', ignoring", parameter.first.c_str());
+      return false;
+    }
+    if (parameter.second.short_desc.empty())
+    {
+      RCLCPP_WARN(get_logger(), "Empty short_desc for '%s', ignoring", parameter.first.c_str());
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void CmdVelMux::configureFromParameters(const std::map<std::string, ParameterValues> & parameters)
 {
-  std::vector<std::shared_ptr<CmdVelSub>> new_list(names.size());
-  for (unsigned int i = 0; i < names.size(); i++)
+  std::map<std::string, std::shared_ptr<CmdVelSub>> new_map;
+
+  for (const std::pair<std::string, ParameterValues> & parameter : parameters)
   {
-    // Parse entries on YAML
-    std::string new_sub_name = names[i];
-    auto old_sub = std::find_if(list_.begin(), list_.end(),
-                                [&new_sub_name](const std::shared_ptr<CmdVelSub>& sub)
-                                                {return sub->name_ == new_sub_name;});
-    if (old_sub != list_.end())
+    const std::string & key = parameter.first;
+    const ParameterValues & parameter_values = parameter.second;
+    // Check if parameter subscriber has all its necessary values
+    if (map_.count(key) != 0)
     {
-      // For names already in the subscribers list, retain current object so we don't re-subscribe to the topic
-      new_list[i] = *old_sub;
+      // For names already in the subscribers map, retain current object so we don't re-subscribe to the topic
+      new_map[key] = map_[key];
     }
     else
     {
-      new_list[i] = std::make_shared<CmdVelSub>();
+      new_map[key] = std::make_shared<CmdVelSub>();
     }
+
     // update existing or new object with the new configuration
 
-    // Fill attributes with a YAML node content
-    double new_timeout;
-    std::string new_topic;
-    new_list[i]->name_ = names[i];
-    new_topic = topics[i];
-    new_timeout = timeouts[i];
-    new_list[i]->priority_ = priorities[i];
-    new_list[i]->short_desc_ = short_descs[i];
+    new_map[key]->name_ = key;
+    new_map[key]->values_.priority = parameter_values.priority;
```

Comment:
@clalancette @Lobotuerk nit^2: you may also do `ParameterValues & new_parameter_values = new_map[key];` for improved readability below.

Reply:
Dont get this one either, could you explain further

Reply:
I meant that instead of using `new_map[key]->values_` all over the place, we could do `ParameterValues & new_parameter_values = new_map[key]`. It's not shorter but it's easier to read.